src/Bicep.Core.UnitTests/Parsing/ParamsParserTests.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Diagnostics;
+using System.Text;
+using Bicep.Core.Syntax;
+using Bicep.Core.UnitTests.Utils;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bicep.Core.UnitTests.Parsing
+{
+    [TestClass]
+    public class ParamsParserTests
+    {
+        [DataTestMethod]
+        [DataRow("true", "true", typeof(BooleanLiteralSyntax))]
+        [DataRow("false", "false", typeof(BooleanLiteralSyntax))]
+        [DataRow("432", "432", typeof(IntegerLiteralSyntax))]
+        [DataRow("1125899906842624", "1125899906842624", typeof(IntegerLiteralSyntax))]
+        [DataRow("null", "null", typeof(NullLiteralSyntax))]
+        [DataRow("'hello world!'", "'hello world!'", typeof(StringSyntax))]
+        public void LiteralExpressionsShouldParseCorrectly(string text, string expected, Type expectedRootType)
+        {
+            RunExpressionTest(text, expected, expectedRootType);
+        }
+
+        [TestMethod]
+        public void TestDeclaringParams()
+        {
+            Trace.WriteLine("Test running!");
+            
+            var paramIntTest = ParamsParserHelper.ParamsParse("param myint = 12 \n");
+            var paramStringTest = ParamsParserHelper.ParamsParse("param mystr = 'hello world' \n");
```

Comment:
We need to verify the output of `ParamsParserHelper.ParamsParse`.

Reply:
The test can also be changed to a [data-driven test](https://www.meziantou.net/mstest-v2-data-tests.htm), e.g.: https://github.com/Azure/bicep/blob/4eea92af52dfc98fdee8f2c4920f442708eb1a38/src/Bicep.Core.UnitTests/Parsing/ParserTests.cs#L33.

Reply:
With the latest iteration, it's fairly simple to change the test to data-driven now:

```C#
[DataTestMethod]
[DataRow("param myint = 12 \n")]
[DataRow("param mystr = 'hello world' \n")]
public void TestParsingParameterAssignment(string text)
{
    var programSyntax = ParamsParserHelper.ParamsParse(text);
    programSyntax.Children.OfType<ParameterAssignmentSyntax>().Should().HaveCount(1);
}
```